### DISCUSSION TITLE
[Bug Report] Conflict between msgpack 1.0.0 and the deprecated msgpack-python

### DISCUSSION THREAD

**Discussion by wangwwno1** [microsoft/AirSim#3511] [Support Q&A] [answered]
## Bug report 

- AirSim Version: a0d4d0e9672900e6182aab5f6cb5b59a09a85509
- UE/Unity version: N/A
- autopilot version: N/A
- OS Version: N/A

### What's the issue you encountered?

> This package is deprecated. Install msgpack instead.

https://pypi.org/project/msgpack-python/
When users install both msgpack and airsim they will have both **msgpack** and **msgpack-python** installed, which conflict.
Edit: After further examination, I find that this issue is related to the `msgpack-rpc-python`.
They have a PR to fix this msgpack-rpc/msgpack-rpc-python#23 but by 2020.12.18 it hasn't merge yet.

### Settings

N/A

### How can the issue be reproduced?

1. Install airsim, then install **msgpack** by `pip install msgpack==1.0.2` Note:  install **msgpack, NOT msgpack-python**, and msgpack must be newer than 1.0.0 as they change the api in this version. 
2. Run the code below and you will get an error
```
import airsim
client = airsim.MultirotorClient()
client.ping()
```
### Include full error message in text form

```
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/home/USER/anaconda3/envs/VeriFC/lib/python3.8/site-packages/airsim/client.py", line 44, in ping
    return self.client.call("ping")
  File "/home/USER/anaconda3/envs/VeriFC/lib/python3.8/site-packages/msgpackrpc/session.py", line 41, in call
    return self.send_request(method, args).get()
  File "/home/USER/anaconda3/envs/VeriFC/lib/python3.8/site-packages/msgpackrpc/session.py", line 51, in send_request
    self._transport.send_message([message.REQUEST, msgid, method, args])
  File "/home/USER/anaconda3/envs/VeriFC/lib/python3.8/site-packages/msgpackrpc/transport/tcp.py", line 89, in send_message
    self.connect()
  File "/home/USER/anaconda3/envs/VeriFC/lib/python3.8/site-packages/msgpackrpc/transport/tcp.py", line 98, in connect
    socket = ClientSocket(stream, self, self._encodings)
  File "/home/USER/anaconda3/envs/VeriFC/lib/python3.8/site-packages/msgpackrpc/transport/tcp.py", line 53, in __init__
    BaseSocket.__init__(self, stream, encodings)
  File "/home/USER/anaconda3/envs/VeriFC/lib/python3.8/site-packages/msgpackrpc/transport/tcp.py", line 12, in __init__
    self._packer = msgpack.Packer(encoding=encodings[0], default=lambda x: x.to_msgpack())
  File "msgpack/_packer.pyx", line 124, in msgpack._cmsgpack.Packer.__init__
TypeError: __init__() got an unexpected keyword argument 'encoding'
```

**What's better than filing an issue? Filing a pull request :).**

**Comment by ahmed-elsaharti** [comment]
@wangwwno1 I tested this on a new machine and everything worked as expected when following [the docs](https://microsoft.github.io/AirSim/apis/) and installing `msgpack-rpc-python`
![image](https://user-images.githubusercontent.com/49626509/103193534-5c936f00-48a2-11eb-8e15-92e18bf7961b.png)

The error did appear when i tried to install `msgpack`. But its not what the docs recommend anyway.

Edit: To fix this, uninstall the <del>airsim,</del> msgpack, msgpack-python and the msgpack-rpc-python packages through:
<del>`pip uninstall airsim`</del>
`pip uninstall msgpack`
`pip uninstall msgpack-python`
`pip uninstall msgpack-rpc-python`
then follow [the docs](https://microsoft.github.io/AirSim/apis/) and reinstall the `airsim `and `msgpack-rpc-python` packages

**Comment by wangwwno1** [comment]
@ahmed-elsaharti 
Please install msgpack by `pip install msgpack==1.0.2` and try again.
`msgpack` and `msgpack-python` are two different packages, and the latter one is deprecated.

**Comment by ahmed-elsaharti** [comment]
@wangwwno1 it shouldn't matter (as far as the scope of Airsim goes) if either of those packages produces errors though as long as the one recommended in the docs works.

If you're trying to make this work just for airsim i would recommend uninstalling the packages mentioned above and installing `msgpack-rpc-python` which should work.

I'll also do some testing on one more machine to make sure the results are consistent.

**Comment by wangwwno1** [comment, marked as answer]
@ahmed-elsaharti 
Thanks for your suggestion, but I need the [Ray](https://github.com/ray-project/ray) library to parallelize the data collection process and `msgpack-python` is conflict with it. I resolve this issue by:
1. install msgpack-rpc-python in this PR msgpack-rpc/msgpack-rpc-python#23 and
2. Remove argument `pack_encoding="utf-8"` and `unpack_encoding="utf-8"` in `VehicleClient`, which locatetd in airsim/client.py

Since msgpack v1.0.0 they have set the default encoding to utf-8 so these arguments are no longer effective.
